src/banking_stage.rs

```diff
@@ -43,26 +42,39 @@ impl BankingStage {
         let thread_hdl = Builder::new()
             .name("solana-banking-stage".to_string())
             .spawn(move || {
-                let (hash_sender, hash_receiver) = channel();
-                let (poh_service, poh_receiver) =
-                    PohService::new(bank.last_id(), hash_receiver, tick_duration);
+                let poh_service = PohService::new(bank.last_id(), tick_duration.is_some());
+
+                let mut last_tick = Instant::now();
+
                 loop {
+                    let timeout =
+                        tick_duration.map(|duration| duration - (Instant::now() - last_tick));
+
                     if let Err(e) = Self::process_packets(
+                        timeout,
                         &bank,
-                        &hash_sender,
-                        &poh_receiver,
+                        &poh_service,
                         &verified_receiver,
                         &entry_sender,
                     ) {
                         match e {
                             Error::RecvTimeoutError(RecvTimeoutError::Disconnected) => break,
                             Error::RecvTimeoutError(RecvTimeoutError::Timeout) => (),
+                            Error::RecvError(_) => break,
```

Comment:
Is this an artifact of the previous strategy or still relevant?

Reply:
I have a blocking recv() in process_packets()